nsv13/code/modules/overmap/starmap.dm

```diff
@@ -0,0 +1,152 @@
+#define SHIPINFO 0
+#define STARMAP 1
+
+/**
+	Starmap code largely copied from FT13's starmap, so credit to them! (And monster860...again)
+*/
+
+
+/datum/asset/simple/starmap
+	assets = list(
+		"space.png" = 'nsv13/icons/assets/space.png')
+
+/obj/machinery/computer/ship/navigation
+	name = "FTL Navigation console"
+	desc = "A computer which can interface with the FTL drive to allow the ship to travel vast distances in space."
+	icon_screen = "ftl"
+	var/datum/star_system/selected_system = null
+	var/screen = STARMAP
+
+/obj/machinery/computer/ship/navigation/attack_hand(mob/user)
+	ui_interact(user)
+
+/obj/machinery/computer/ship/navigation/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state) // Remember to use the appropriate state.
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		var/datum/asset/assets = get_asset_datum(/datum/asset/simple/starmap)
+		assets.send(user)
+		ui = new(user, src, ui_key, "starmap", name, 800, 660, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/ship/navigation/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
+	if(!has_overmap())
+		return
+	switch(action)
+		if("map")
+			screen = STARMAP
+		if("shipinf")
+			screen = SHIPINFO
+		if("select_system")
+			var/datum/star_system/target = locate(params["star_id"])
+			if(!istype(target))
+				return
+			selected_system = target
+			screen = 2
+			. = TRUE
+		if("jump")
+			linked.ftl_drive.jump(selected_system)
+			. = TRUE
+
+/obj/machinery/computer/ship/navigation/ui_data(mob/user)
+	if(!has_overmap())
+		return
+	var/list/data = list()
+	var/list/info = SSstar_system.ships[linked]
+	var/list/lines = list()
+	var/datum/star_system/current_system = info["current_system"]
+	SSstar_system.update_pos(linked)
+
+	if(screen == 0) // ship information
+		if(linked.ftl_drive)
+			data["ftl_progress"] = linked.ftl_drive.progress
+			data["ftl_goal"] = linked.ftl_drive.spoolup_time //TODO
+		var/datum/star_system/target_system = info["target_system"]
+		if(!target_system)
+			data["in_transit"] = FALSE
+			data["star_id"] = "\ref[current_system]"
+			data["star_name"] = current_system.name
+		else
+			data["in_transit"] = TRUE
+			data["from_star_id"] = "\ref[info["last_system"]]]"
+			data["from_star_name"] = info["last_system"].name
+			data["to_star_id"] = "\ref[target_system]"
+			data["to_star_name"] = target_system.name
+			data["time_left"] = max(0, (info["to_time"] - world.time) / 1 MINUTES)
+
+	else if(screen == 1) // star system map screen thing
+
+		var/list/systems_list = list()
+		if(info["current_system"])
+			data["focus_x"] = info["current_system"].x
+			data["focus_y"] = info["current_system"].y
+		else
+			data["focus_x"] = info["x"]
+			data["focus_y"] = info["y"]
+		for(var/datum/star_system/system in SSstar_system.systems) // for each system
```

Comment:
This proc area seems rather inefficient. But I dont know TGUI so its probably suitable for now.